lib/spring/application.rb

```diff
@@ -319,7 +319,10 @@ def shush_backtraces
             ensure
               if $!
                 lib = File.expand_path("..", __FILE__)
-                $!.backtrace.reject! { |line| line.start_with?(lib) } unless $!.backtrace.frozen?
+                unless $!.backtrace.frozen?
+                  $!.backtrace.reject! { |line| line.start_with?(lib) }
+                  $!.backtrace_locations.reject! { |line| line.path.start_with?(lib) }
```

Comment:
```suggestion
                  $!.backtrace_locations.reject! { |line| line.path&.start_with?(lib) }
```

Pretty sure `path` can be `nil` if the method is implemented in C.

Reply:
Also, while it's very convoluted, `.backtrace` could be mutable, but `.backtrace_location` frozen.